File: features/manager/services/auth.service.ts
import api from "@/common/lib/axios";
import { ApiResponse } from "@/common/types/api.types";
import { LoginRequest, LoginResponse } from "../types/auth.types";

export const login = async (request: LoginRequest) => {
    const response = await api.post<ApiResponse<LoginResponse>>("/auth/login", request);
    return response.data;
}

export const logout = async () => {
    const response = await api.post<ApiResponse<null>>("/auth/logout");
    return response.data;
}

File: apps/backend/src/main/java/com/supertech/superbatch/manager/auth/service/AuthService.java
package com.supertech.superbatch.manager.auth.service;

import com.supertech.superbatch.manager.auth.dto.LoginRequest;
import com.supertech.superbatch.manager.auth.dto.LoginResponse;

public interface AuthService {
    LoginResponse login(LoginRequest request);

    void logout();
}
